File: FoodOrganizer/app/src/main/java/es/fron99/foodorganize/Adapters/AdapterListMenus.java
package es.fron99.foodorganize.Adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.PopupMenu;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import es.fron99.foodorganize.Models.Menu;
import es.fron99.foodorganize.R;

public class AdapterListMenus extends RecyclerView.Adapter<AdapterListMenus.ViewHolder>{

    private ArrayList<Menu> menus;

    public static class ViewHolder extends RecyclerView.ViewHolder {
        private final TextView txtViewNameMenu, txtViewSmallDescriptionMenu;
        private final ImageView imageView, optionsMenu;

        public ViewHolder(View view) {
            super(view);
            txtViewNameMenu = view.findViewById(R.id.nameFood);
            txtViewSmallDescriptionMenu = view.findViewById(R.id.smallDescriptionFood);
            imageView = view.findViewById(R.id.imageView);
            optionsMenu = view.findViewById(R.id.textViewOptions);
        }

        public TextView getTxtViewNameMenu() {
            return txtViewNameMenu;
        }

        public TextView getTxtViewSmallDescriptionMenu() {
            return txtViewSmallDescriptionMenu;
        }

        public void setTxtViewNameMenu(String nameMenu) {
            txtViewNameMenu.setText(nameMenu);
        }

        public void setTxtViewSmallDescriptionMenu(String smallDescriptionMenu) {
            txtViewSmallDescriptionMenu.setText(smallDescriptionMenu);
        }

        public ImageView getImageView() {
            return imageView;
        }

        public void setImageView(int resourseId) {
            imageView.setImageResource(resourseId);
        }

        public ImageView getOptionsMenu() {
            return optionsMenu;
        }

        public void setOptionsMenu(int id) {
            optionsMenu.setImageResource(id);
        }

    }

    public AdapterListMenus(ArrayList<Menu> dataSet) {
        menus = new ArrayList<>(dataSet);
    }

    @NonNull
    @Override
    public AdapterListMenus.ViewHolder onCreateViewHolder(ViewGroup viewGroup, int viewType) {
        View view = LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.row_list_food_and_menus, viewGroup, false);
        return new AdapterListMenus.ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(AdapterListMenus.ViewHolder viewHolder, final int position) {
        Menu menu = this.menus.get(position);
        viewHolder.setTxtViewNameMenu(menu.getName());
        viewHolder.setTxtViewSmallDescriptionMenu(menu.getSmallDescription());
        viewHolder.setImageView(R.drawable.icon_menus);
        viewHolder.getOptionsMenu().setOnClickListener(view -> {
            PopupMenu popup = new PopupMenu(view.getContext(), viewHolder.getOptionsMenu());
            popup.inflate(R.menu.menu_row_list_menu);
            popup.setOnMenuItemClickListener(item -> {
                switch (item.getItemId()) {
                    case R.id.itemEliminar:
                        //handle menu1 click
                        return true;
                    case R.id.itemModificar:
                        //handle menu2 click
                        return true;
                    default:
                        return false;
                }
            });
            popup.show();

        });
    }

    @Override
    public int getItemCount() {
        return menus.size();
    }

}

File: FoodOrganizer/app/src/main/java/es/fron99/Foodorganize/Adapters/AdapterListMenus.kt
package es.fron99.Foodorganize.Adapters

import android.view.LayoutInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.PopupMenu
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import es.fron99.Foodorganize.Models.Menu
import es.fron99.Foodorganize.Models.TimeMenu
import es.fron99.Foodorganize.R
import java.util.*


class AdapterListMenus(dataSet: ArrayList<Menu>?) : RecyclerView.Adapter<AdapterListMenus.ViewHolder>() {
    private var menus: ArrayList<Menu> = ArrayList(dataSet)

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val txtViewNameMenu: TextView = view.findViewById(R.id.nameFood)
        val txtViewSmallDescriptionMenu: TextView = view.findViewById(R.id.smallDescriptionFood)
        val imageView: ImageView = view.findViewById(R.id.imageView)
        val optionsMenu: ImageView = view.findViewById(R.id.textViewOptions)

        fun setTxtViewNameMenu(nameMenu: String?) {
            txtViewNameMenu.text = nameMenu
        }

        fun setTxtViewSmallDescriptionMenu(smallDescriptionMenu: String?) {
            txtViewSmallDescriptionMenu.text = smallDescriptionMenu
        }

        fun setImageView(resourseId: Int) {
            imageView.setImageResource(resourseId)
        }

        fun setOptionsMenu(id: Int) {
            optionsMenu.setImageResource(id)
        }

    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(viewGroup.context)
                .inflate(R.layout.row_list_food_and_menus, viewGroup, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, position: Int) {
        val menu = menus[position]
        viewHolder.setTxtViewNameMenu(menu.name)
        viewHolder.setTxtViewSmallDescriptionMenu(menu.smallDescription)
        viewHolder.setImageView(R.drawable.icon_menus)
        viewHolder.optionsMenu.setOnClickListener { view: View ->
            val popup = PopupMenu(view.context, viewHolder.optionsMenu)
            popup.inflate(R.menu.menu_row_list_menu)
            popup.setOnMenuItemClickListener { item: MenuItem ->
                when (item.itemId) {
                    R.id.itemEliminar ->                         //handle menu1 click
                        return@setOnMenuItemClickListener true
                    R.id.itemModificar ->                         //handle menu2 click
                        return@setOnMenuItemClickListener true
                    else -> return@setOnMenuItemClickListener false
                }
            }
            popup.show()
        }
    }

    override fun getItemCount(): Int {
        return menus.size
    }

    fun changeData(dataSet: ArrayList<Menu>?){
        this.menus.clear()
        this.menus = ArrayList(dataSet)
        this.notifyDataSetChanged()
    }

}